include/boost/python/numpy/dtype.hpp

```diff
@@ -70,37 +70,37 @@ class BOOST_NUMPY_DECL dtype : public object {
 
 };
 
-bool equivalent(dtype const & a, dtype const & b);
+BOOST_NUMPY_DECL bool equivalent(dtype const & a, dtype const & b);
 
 namespace detail
 {
 
```

Comment:
Sorry for the delay. I seem to have missed the notification about your comment.

Anyways, the most pertinent change is the sprinkling of several "BOOST_NUMPY_DECL" statements to force the generation of still missing symbols and objects.

The other changes were necessitated by compilation problems. Under MSVC there were several chicken-or-the-egg -problems with the caused need to apply BOOST_NUMPY_DECL which was problematic due to the forward declarations and such. Also the whole dtype engine seemed to be rather fragile and caused semi-random segfaults depending on the builds. 

After a moderate diagnosis effort I noticed that if you re-arrange the code and especially the declarations in certain logical order the compiler is happy and you can simplify the code and get rid of the segfaults. So yes, apart from BOOST_NUMPY_DECL, the patch is mostly re-arranging the code, but it is done to keep compiler happy.

I've been running this code under several applications and extreme load. As far as I can tell the DLLs produced appear to be rock solid.